mslice/models/cut/matplotlib_cut_plotter.py

```diff
@@ -33,35 +30,43 @@ def plot_cut_from_xye(self, x, y, e, x_units, selected_workspace, plot_over, leg
         plt.xlabel(self._getDisplayName(x_units, self._cut_algorithm.getComment(selected_workspace)), picker=picker)
         plt.ylabel(INTENSITY_LABEL, picker=picker)
         plt.autoscale()
+        # Note that we cannot cache the reference to canvas and axes here because under the keep/make current mechanism
+        # there is no one-to-one link between any particular CutPlotter and any CutPlot (or mpl canvas) object
+        plt.gcf().canvas.manager.add_cut_plot(self)
         if not plot_over:
             plt.gcf().canvas.manager.update_grid()
-        self.axes.draw_artist(self.canvas.figure.get_children()[1])
-        self.canvas.blit(self.canvas.figure.gca().clipbox)
+        if self.background is None:
+            self._create_cut()
+        plt.gcf().canvas.restore_region(self.background)
+        try:
+            plt.gca().draw_artist(plt.gcf().canvas.figure.get_children()[1])
+        except AttributeError:
+            plt.gcf().canvas.draw()
+        plt.gcf().canvas.blit(plt.gcf().canvas.figure.gca().clipbox)
```

Comment:
This is better, but in the case where `draw()` is called, we shouldn't need to call `blit` afterwards.